test/k23si/QueryTest.cpp

```diff
@@ -251,10 +252,46 @@ seastar::future<> runScenario01() {
     });
 }
 
-// Query projection cases
+// Simple reverse scan queries with no filter or projection
 seastar::future<> runScenario02() {
```

Comment:
A few points on the tests:
- Adding a test document like docs/RFC/K23SI_testing.md would be helpful here to make sure we cover all the corner cases
- I think at least one of the tests should check the order of the results to see if they are in fact in reverse order
- Let's add a TODO to add test cases when we are capable of changing the partition map